k8s/k8s.go

```diff
@@ -0,0 +1,207 @@
+		return nil, err
+	}
+	defer response.Body.Close()
+	return ParseVulnDBData(db)
+}
+
+const (
+	// excludeNonCoreComponentsCves exclude  cves with missing data or non k8s core components
+	excludeNonCoreComponentsCves = "CVE-2019-11255,CVE-2020-10749,CVE-2020-8554"
+)
+
+func Update() error {
+	if err := update(); err != nil {
+		return xerrors.Errorf("error in k8s update: %w", err)
+	}
+	return nil
+}
+
+func update() error {
+	log.Printf("Fetching k8s cves")
+	k8sdb, err := Collect()
+	if err != nil {
+		return err
+	}
+	for _, cve := range k8sdb.Cves {
+		if err = uu.Write(filepath.Join(uu.VulnListDir(), "upstream", fmt.Sprintf("%s.json", cve.ID)), cve); err != nil {
+			return xerrors.Errorf("failed to save k8s CVE detail: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func ParseVulnDBData(db CVE) (*VulnDB, error) {
+	var fullVulnerabilities []*osv.OSV
+	for _, item := range db.Items {
+		if strings.Contains(excludeNonCoreComponentsCves, item.ID) {
+			continue
+		}
+		for _, cveID := range getMultiIDs(item.ID) {
+			vulnerability, err := parseMitreCve(item.ExternalURL, cveID)
+			if err != nil {
+				return nil, err
+			}
+			if cveMissingImportantData(vulnerability) {
+				continue
+			}
+			descComponent := getComponentFromDescription(item.ContentText, vulnerability.Package)
+			fullVulnerabilities = append(fullVulnerabilities, &osv.OSV{
+				ID:         cveID,
+				Modified:   item.DatePublished,
+				Published:  item.DatePublished,
+				Summary:    item.Summary,
+				Details:    vulnerability.Description,
+				Affected:   getAffectedEvents(vulnerability.versions, getComponentName(descComponent, vulnerability.Package), vulnerability.CvssV3),
+				References: []osv.Reference{{Url: item.URL}, {Url: item.ExternalURL}},
+			})
+		}
+		err := validateCvesData(fullVulnerabilities)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &VulnDB{fullVulnerabilities}, nil
+}
+
+func getAffectedEvents(v []*Version, p string, cvss Cvssv3) []osv.Affected {
+	affected := make([]osv.Affected, 0)
+	for _, av := range v {
+		if len(av.Introduced) == 0 {
+			continue
+		}
+		if av.Introduced == "0.0.0" {
+			av.Introduced = "0"
+		}
+		events := make([]osv.Event, 0)
+		ranges := make([]osv.Range, 0)
+		if len(av.Introduced) > 0 {
+			events = append(events, osv.Event{Introduced: av.Introduced})
+		}
+		if len(av.Fixed) > 0 {
+			events = append(events, osv.Event{Fixed: av.Fixed})
+		} else if len(av.LastAffected) > 0 && len(av.Fixed) == 0 {
+			events = append(events, osv.Event{LastAffected: av.LastAffected})
+		}
+		if len(av.Introduced) > 0 && len(av.LastAffected) == 0 && len(av.Fixed) == 0 {
+			events = append(events, osv.Event{LastAffected: av.Introduced})
+		}
+		ranges = append(ranges, osv.Range{
+			Events: events,
+		})
+		affected = append(affected, osv.Affected{Ranges: ranges, Package: osv.Package{Name: p, Ecosystem: "k8s"}, Severities: []osv.Severity{{Type: cvss.Vector, Score: fmt.Sprintf("%.1f", cvss.Score)}}})
```

Comment:
I'm not sure, but "Kubernetes" might be better.
https://ossf.github.io/osv-schema/#affectedpackage-field

```suggestion
		affected = append(affected, osv.Affected{Ranges: ranges, Package: osv.Package{Name: p, Ecosystem: "Kubernetes"}, Severities: []osv.Severity{{Type: cvss.Vector, Score: fmt.Sprintf("%.1f", cvss.Score)}}})
```